### PR TITLE
perf(decide): Speed up decide endpoint by shortcircuiting middleware

### PR DESCRIPTION
**Pull request by timgl**
## Problem

I suspect our middleware is adding ~100ms to /decide calls 

## Changes

- Shortcut most of our middleware when calling /decide
- Get rid of `is_authenticated` in the decide response, as this only works on posthog.com (or if people self host on the same domain name)

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

unit tests

### CODE REVIEW

File: posthog/settings/web.py [hunk 1]
```diff
@@ -47,10 +47,11 @@
     "django_structlog.middlewares.RequestMiddleware",
     "django_structlog.middlewares.CeleryMiddleware",
     "django.middleware.security.SecurityMiddleware",
-    "posthog.middleware.AllowIPMiddleware",
+    "posthog.middleware.ShortCircuitMiddleware",
```
Comment (guidoiaquinti) [inline posthog/settings/web.py:50]: Overall I'm not a big fan of this approach as we are silently short circuiting all the middlewares below this line using a condition in the middleware itself. 

`/decide` is pretty simple endpoint to audit but we might be missing an application behaviour that can depends on one of middlewares below.

**Review by guidoiaquinti — APPROVED** [review]
Non blocking comment. Otherwise, LGTM

File: posthog/settings/web.py (hunk 1 above)
Comment (timgl) [reply posthog/settings/web.py:50]: Yeah definitely. I want to just test my hypothesis here, if it does work I want to apply this to the capture endpoint as well so I'll spend a bit more time

Comment (guidoiaquinti) [reply posthog/settings/web.py:50]: Actually after reviewing this further this is a pattern we already use in our codebase https://github.com/PostHog/posthog/blob/46faa8fab258b04cf4aa6fa27fbe39563bb998d2/posthog/health.py#L237-L253

Comment (timgl) [reply posthog/settings/web.py:50]: I'll put this live and see if it makes a difference. If it does I'll refactor all this into something more sensible

File: posthog/api/test/test_decide.py [hunk 2]
```diff
@@ -55,31 +55,6 @@ def test_defaults_to_v2_if_conflicting_parameters(self):
...
-        user.toolbar_mode = "disabled"
-        user.save()
-
-        self.team.app_urls = ["https://example.com/maybesubdomain"]
-        self.team.save()
-
-        # Make sure the endpoint works with and without the trailing slash
-        response = self.client.get("/decide", HTTP_ORIGIN="https://example.com").json()
-        self.assertEqual(response["isAuthenticated"], True)
-        self.assertIsNone(response["editorParams"].get("toolbarVersion"))
```
Comment (Twixes) [inline posthog/api/test/test_decide.py:81]: Why were those tests removed? It seems they signaled an actual regression.

Comment (timgl) [reply posthog/api/test/test_decide.py:81]: (said in slack but) we purposefully removed this as some people complained the toolbar always being there was annoying + it slowed /decide down. You can still launch toolbar from the app. @tiina303 we should highlight this change in the release

**Pull request merged**